c2rust-analyze/src/dataflow/type_check.rs

```diff
@@ -48,16 +49,25 @@ impl<'tcx> TypeChecker<'tcx, '_> {
     }
 
     pub fn visit_place_ref(&mut self, pl: PlaceRef<'tcx>, ctx: PlaceContext) -> LTy<'tcx> {
+        let mutbl = match ctx {
+            PlaceContext::NonMutatingUse(..) => Some(Mutability::Not),
+            PlaceContext::MutatingUse(..) => Some(Mutability::Mut),
+            PlaceContext::NonUse(..) => None,
+        };
+
         let mut lty = self.acx.local_tys[pl.local];
         let mut prev_deref_ptr = None;
 
         for proj in pl.projection {
             match proj {
                 ProjectionElem::Deref => {
                     // All derefs except the last are loads, to retrieve the pointer for the next
-                    // deref.  The last deref is either a load or a store, depending on `ctx`.
+                    // deref.  However, if the last deref is `&mut` (and is used mutably), then the
```

Comment:
is the last deref being &mut and used mutably determined exclusively by `mutbl`?

Reply:
Yes - I updated the comment slightly